### PR TITLE
[greenboot] seperate  logs to a standalone plugin.

### PR DESCRIPTION
**Pull request by eslutsky**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [ ] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/greenboot.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Greenboot(Plugin, RedHatPlugin):
+    """The greenboot plugin collects systemd service logs and configuration.
+    """
+
+    short_desc = 'Greenboot'
+    plugin_name = 'greenboot'
+    plugin_timeout = 300
+    packages = ('microshift-greenboot')
```
Comment (arif-ali) [inline sos/report/plugins/greenboot.py:20]: you need to add a comma after `'microshift-greenboot'`, as this needs to be a tuple

**Review by arif-ali — CHANGES_REQUESTED** [review]

**Review by arif-ali — APPROVED** [review]

File: sos/report/plugins/greenboot.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+class Greenboot(Plugin, RedHatPlugin):
+    """The greenboot plugin collects systemd service logs and configuration.
+    """
+
+    short_desc = 'Greenboot'
+    plugin_name = 'greenboot'
+    plugin_timeout = 300
```
Comment (pmoravec) [inline sos/report/plugins/greenboot.py:19]: This is the default value (cf. https://github.com/sosreport/sos/blob/main/sos/report/plugins/__init__.py#L534 and https://github.com/sosreport/sos/blob/main/sos/utilities.py#L47), so a redundant setting..?

**Comment by pmoravec** [comment]
Commit message speaks about journal logs, but the change would collect just a config file - do I miss something..?

**Comment by bmr-cymru** [comment]
Commit message for a plugin wants to be something like `[greenboot] separate logs to a standalone plugin`.

File: sos/report/plugins/greenboot.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    packages = ('microshift-greenboot',)
```
Comment (dhellmann) [inline sos/report/plugins/greenboot.py:20]: We probably don't want to tie this to MicroShift.

Comment (eslutsky) [reply sos/report/plugins/greenboot.py:20]: @dhellmann , greenboot is distributed as `microshift-greenboot` rpm, are we going to change that also?

**Review by TurboTurtle — COMMENTED** [review]
LGTM, however this needs to be rebased on current `main` as there is now a merge conflict based upon another change to the `microshift` plugin (minor change to the services tuple).

Once that's done, this should be good. Note 4.6.1 is scheduled to close tomorrow.

**Comment by eslutsky** [comment]
> LGTM, however this needs to be rebased on current `main` as there is now a merge conflict based upon another change to the `microshift` plugin (minor change to the services tuple).
> 
> Once that's done, this should be good. Note 4.6.1 is scheduled to close tomorrow.

ACK and done .

File: sos/report/plugins/greenboot.py (hunk 3 above)
Comment (dhellmann) [reply sos/report/plugins/greenboot.py]: The `microshift-greenboot` RPM contains the scripts for MicroShift to integrate with greenboot, but the actual system software is in the `greenboot` RPM.

**Review by arif-ali — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**